src/configuration.ts

```diff
@@ -142,6 +141,35 @@ export function findConfigurationPath(suppliedConfigFilePath: string | null, inp
 }
 
 /**
+ * Find a file by name in a directory or any ancestory directory.
+ * This is case-insensitive, so it can find 'TsLiNt.JsOn' when searching for 'tslint.json'.
+ */
+function findup(filename: string, directory: string): string | undefined {
+    while (true) {
+        const res = findFile(directory);
+        if (res) {
+            return res;
+        }
+
+        const parent = path.dirname(directory);
+        if (parent === directory) {
+            return undefined;
+        }
+        directory = parent;
+    }
+
+    function findFile(cwd: string): string | undefined {
+        if (fs.existsSync(path.join(cwd, filename))) {
+            return filename;
+        }
+
+        // Try reading in the entire directory and looking for a file with different casing.
```

Comment:
maybe we should deprecate that behavior. It's already inconsistent with searching in home directory which is case sensitive.